scripts/lint-samples.js

```diff
@@ -0,0 +1,44 @@
+const { execSync } = require('child_process');
+const samples = require('./samples.json');
+
+/**
+ * Start linting process only for the samples that were affected by the new changes in:
+ * - create-sitecore-jss/src/templates/**
+ * - create-sitecore-jss/src/initializers/**
+ */
+
+const affectedTemplates = execSync('git diff --name-only packages/create-sitecore-jss', {
+  encoding: 'utf-8',
+})
+  .split('\n')
+  .map((filepath) => {
+    // Extracting template names from the filepath
+    const template = filepath.match(/(templates|initializers)\/([^\/]*)/);
```

Comment:
Looks like this approach would miss changes made in src/common or src/<root> files. Though, I don't know if that's a huge concern? Those changes would be caught in our separate automated tests suite.

Reply:
You are right, I think that we need to consider more specific changes, that are directly related to the template
rest of them will be caught